### PR TITLE
Adding last seen idle time in SessionState

### PR DESCRIPTION
**Pull request by foyzur**
This PR addresses external tools such as WLM requirement to determine the idle time for each session. We add a per-session last seen idle time across all QEs on each segment. Each QE updates this value in its own SessionState whenever it becomes idle. To determine how long a session has been idle we can run a query like this:

```
frahman=# select sess_id, now() - last_idle_time as idle_duration from session_state.session_level_memory_consumption where active_qe_count = 0 and segid = -1;
 sess_id |  idle_duration
---------+-----------------
      16 | 00:01:45.882067
(1 row)
```

While the value is mostly useful on the master, the operation itself is not very expensive and so we record it on all the segments. Note: WLM currently issues cancellation on the segment directly (in utility mode). Therefore, exposing segment level idle timeout might offer additional flexibility to define WLM rules. This can also be coupled with "idle in transaction" vs. true idle using pg_stat_activity (or we can expose the idle in transaction status in the future, if necessary).

### CODE REVIEW

**Comment by foyzur** [comment]
@karthijrk @hsyuan @hardikar @greenplum-db/gp-workload-manager please take a look.

File: src/bin/gp_session_state/gp_session_state.sql.in [hunk 1]
```diff
@@ -58,7 +59,8 @@ WITH all_entries AS (
             active_qe_count int, 
             dirty_qe_count int,
             runaway_vmem_mb int, 
-            runaway_command_cnt int
+            runaway_command_cnt int,
+            last_idle_time timestamp with time zone
```
Comment (benchristel) [inline src/bin/gp_session_state/gp_session_state.sql.in]: Could we call this column`idle_start` to parallel `query_start` and `xact_start` in `pg_stat_activity`? To me, `last_idle_time` sounds like it's the most recent time the session was observed to be idle—i.e. the time at which it *stopped* being idle.

— Ben, @kaknikhil, @amilkh

**Comment by kaknikhil** [comment]
Apart from @benchristel's comment , looks good

**Comment by ivannovick** [comment]
quite useful thank you!  Now its easy to see exactly how long a session has been idle

File: src/bin/gp_session_state/gp_session_state.sql.in (hunk 1 above)
Comment (foyzur) [inline src/bin/gp_session_state/gp_session_state.sql.in]: Of course we can. Any comment from the PMs as this is a user facing change? @ozbasarir @vraghavan78 @ivannovick

**Comment by foyzur** [comment]
@benchristel please take a look. I changed all the last_idle_time to idle_start.

**Comment by ozbasarir** [comment]
That looks great. Thank you!

**Comment by benchristel** [comment]
@foyzur looks good to me!

Ben + Amil

**Pull request merged**